modules/payment/commerce_payment.module

```diff
@@ -125,12 +139,16 @@ function commerce_payment_create_field_definitions(array $definitions) {
  *
  * @param array $definitions
  *   The field definitions.
+ * @param bool $delete_storage
+ *   If set TRUE, field storage definition will be deleted also
```

Comment:
I guess it would reference a non-existent payment gateway
The scope of this PR was just to fix the field functionality associated with payments, but I agree that the above needs to be addressed at some point